packages/ra-ui-materialui/src/input/AutocompleteInput.stories.tsx

```diff
@@ -326,61 +336,171 @@ export const OnCreateSlow = () => (
+                            autoFocus
+                        />
+                        <TextField
+                            name="language"
+                            label="The author language"
+                            value={language}
+                            onChange={event => setLanguage(event.target.value)}
+                            autoFocus
+                        />
+                    </Stack>
+                </DialogContent>
+                <DialogActions>
+                    <Button type="submit">Save</Button>
+                    <Button onClick={onCancel}>Cancel</Button>
+                </DialogActions>
+            </form>
+        </Dialog>
+    );
+};
+
+const CreateDialogInput = () => {
+    const [choices, setChoices] = useState(choicesForCreationSupport);
+    return (
+        <AutocompleteInput
+            source="author"
+            choices={choices}
+            create={
+                <CreateAuthorLocal choices={choices} setChoices={setChoices} />
+            }
+            TextFieldProps={{
+                placeholder: 'Start typing to create a new item',
+            }}
+        />
+    );
+};
+
+export const CreateDialog = () => (
     <Wrapper>
+        <CreateDialogInput />
+    </Wrapper>
+);
+
+const CreateLabelInput = () => {
+    const [choices, setChoices] = useState(choicesForCreationSupport);
+    return (
         <AutocompleteInput
             source="author"
-            choices={[
-                { id: 1, name: 'Leo Tolstoy' },
-                { id: 2, name: 'Victor Hugo' },
-                { id: 3, name: 'William Shakespeare' },
-                { id: 4, name: 'Charles Baudelaire' },
-                { id: 5, name: 'Marcel Proust' },
-            ]}
-            onCreate={filter => {
-                const newAuthorName = window.prompt(
-                    'Enter a new author',
-                    filter
-                );
+            choices={choices}
+            onCreate={async filter => {
+                if (!filter) return;
 
-                if (newAuthorName) {
-                    const newAuthor = {
-                        id: choicesForCreationSupport.length + 1,
-                        name: newAuthorName,
-                    };
-                    choicesForCreationSupport.push(newAuthor);
-                    return newAuthor;
-                }
+                const newOption = {
+                    id: choices.length + 1,
+                    name: filter,
+                };
+                setChoices(options => [...options, newOption]);
+                // Wait until next tick to give some time for React to update the state
+                await new Promise(resolve => setTimeout(resolve));
+                return newOption;
             }}
             createLabel="Start typing to create a new item"
         />
+    );
+};
+
+export const CreateLabel = () => (
+    <Wrapper>
+        <CreateLabelInput />
+    </Wrapper>
+);
```

Comment:
Found an issue in this story, where the Autocomplete displays the Create option for an item that already exists

![2024-10-11_14-16](https://github.com/user-attachments/assets/c308f59d-977f-4c0b-8707-f6846d1261d5)